springtime.py

```diff
@@ -0,0 +1,196 @@
+        dataset : str
+            Name of the test dataset
+
+        phenophase : str
+            Name of the phenophase. Either 'budburst','flowers', 'colored_leaves',
+            or 'all'.
+
+    Returns:
+        obs, predictors : tuple
+            Pandas dataframes of phenology observations
+            and associated temperatures.
+    """
+    return pyPhenology.utils.load_test_data(name=dataset, phenophase=phenophase)
+
+
+def split_data(data, method_name):
+    """Split arrays into random train and test subsets.
+
+    Methods are available with sklearn package.
+    Args:
+        dataset : array
+            data to be splitted.
+
+        method_name : str
+            one of the sklearn train test split functions.
+
+    Returns:
+        test, train : tuple
+            arrays for testing and training.
+    """
+    if method_name=="ShuffleSplit":
+        # TODO pass arguments to function
+        rs = ShuffleSplit(n_splits=2, test_size=.25, random_state=2)
+        for train_index, test_index in rs.split(data):
+            return data.iloc[test_index], data.iloc[train_index]
+
+    print(f"Unsupported train_test strategy {method_name}")
+    return None
+            
+
+# metric utilities
+def rmse(obs, pred):
+    """Root mean square error.
+    Args:
+        obs : array
+            array of observations.
+
+        pred : array
+            array of predictions.
+
+    Returns:
+       rmse : float
+            rmse value.
+    """
+    return np.sqrt(((pred - obs)**2).mean())
+
+
+# model utilities
+def check_model_name(model_name):
+    models_source = {
+        "ThermalTime": "pyPhenology.primary_model",
+        "Linear": "pyPhenology.primary_model",
+        "LinearRegression": "sklearn.linear_model",
+    }
+    if models_source.get(model_name):
+        return models_source.get(model_name)
+    raise ValueError(f"Unsupported model {model_name}.")
+
+    
+def select_model(model_name):
+    if check_model_name(model_name) == "pyPhenology.primary_model":
+        Model = pyPhenology.utils.load_model(model_name)
+    elif check_model_name(model_name) == "sklearn.linear_model":
+        Model = getattr(linear_model, model_name)
+    return Model()
+    
+
+# workflow utilities
+def define_metrics(metric_name):
+    if metric_name=="RMSE":
+        return rmse
+    print(f"Unsupported metric method {metric_name}")
+    return None
+
+
+def build_workflow(options, name):
+    # TODO validate_options(options)
+    workflow = {}
+    workflow["observations"], workflow["predictors"] = load_data(
+        options["dataset"], options["phenophase"]
+        )
+
+    workflow["observations_test"], workflow["observations_train"] = split_data(
+        workflow["observations"], options["train_test_strategy"]
+        )
+    
+    workflow["metric_func"] = define_metrics(options["metric_name"])
+    workflow["model"] = select_model(options["model_name"])
+
+    workflow["name"] = name
+    workflow["options"] = options
```

Comment:
It has quite some similarities to existing (but overkill complexity) tools, e.g. https://www.mlflow.org/docs/latest/quickstart.html#using-the-tracking-api

Notice the analogy between how you "save workflow settings", and they "log metrics"